src/components/visx/TimeSeriesChart.tsx

```diff
@@ -169,13 +169,27 @@ export const TimeSeriesChart = <Datum extends {}>({
 
   const [zoomDomainAnimateTo, setZoomDomainAnimateTo] = useState<number | undefined>();
 
+  // handle someone changing defaultZoomDomain
   useEffect(() => {
     if (defaultZoomDomain) {
       const clampedZoomDomain = getClampedZoomDomain(defaultZoomDomain);
       setZoomDomain(clampedZoomDomain);
       setZoomDomainAnimateTo(clampedZoomDomain);
     }
-  }, [defaultZoomDomain, getClampedZoomDomain]);
+  }, [defaultZoomDomain]);
+
+  // Handle data changing somehow
+  useEffect(() => {
+    // hack: if we are zoomed all the way in, it's probably because there was no data before, let's use default
```

Comment:
this is only true onMount right? If a user scrolls to the minZoomDomain I don't think they expect to be corrected to default. That would be a jarring experience.

Reply:
I agree however the current experience is they get corrected any time there's new data and with this change it will ONLY happen when they zoom all the way in.

Reply:
and I can't figure out how to rewrite this component to be better